cmd/serve.go

```diff
@@ -4,151 +4,84 @@
-			updateProbingPeriod = 20 * time.Second
-		}
-		log.Info("Probing interval: ", updateProbingPeriod.String())
-
-		pruneMetricsPeriod, err := time.ParseDuration(cleanMetricsPeriod)
-		if err != nil {
-			log.Warning("Impossible to parse cleaningPeriod value, fallback to 600s")
-			pruneMetricsPeriod = 600 * time.Second
-		}
-		if pruneMetricsPeriod < 240*time.Second {
-			log.Warning("Cleaning Metrics faster than every 4 minutes is not allowed, fallback to 240s")
-			pruneMetricsPeriod = 240 * time.Second
-		}
-		log.Info("Metrics pruning interval: ", pruneMetricsPeriod.String())
-
-		updateDiscoveryTicker := time.NewTicker(updateDiscoveryPeriod)
-		cleanMetricsTicker := time.NewTicker(pruneMetricsPeriod)
-		executeProbingTicker := time.NewTicker(updateProbingPeriod)
-
-		for {
-			select {
-			case <-cleanMetricsTicker.C:
-				log.Info("Cleaning Prometheus metrics for unreferenced nodes")
-				cleanMetrics(esNodesList, allEverKnownEsNodes)
-				cleanMetrics(kibanaNodesList, allEverKnownKibanaNodes)
-
-			case <-updateDiscoveryTicker.C:
-				// Elasticsearch
-				log.Debug("Starting updating ES nodes list")
-				updatedList, err := discoverNodesForService(elasticsearchConsulService)
-				if err != nil {
-					log.Error("Unable to update ES nodes, using last known state")
-					errorsCount.Inc()
-					continue
-				}
-
-				log.Info("Updating ES nodes list")
-				allEverKnownEsNodes = updateEverKnownNodes(allEverKnownEsNodes, updatedList)
-				esNodesList = updatedList
-
-				// Kibana
-				log.Debug("Starting updating Kibana nodes list")
-				kibanaUpdatedList, err := discoverNodesForService(kibanaConsulService)
-				if err != nil {
-					log.Error("Unable to update Kibana nodes, using last known state")
-					errorsCount.Inc()
-					continue
-				}
-
-				log.Info("Updating kibana nodes list")
-				allEverKnownKibanaNodes = updateEverKnownNodes(allEverKnownKibanaNodes, kibanaUpdatedList)
-				kibanaNodesList = kibanaUpdatedList
-
-			case <-executeProbingTicker.C:
-				log.Debug("Starting probing ES nodes")
-
-				sem := new(sync.WaitGroup)
-				for _, node := range esNodesList {
-					sem.Add(1)
-					go func(esNode esnode) {
-						defer sem.Done()
-						probeElasticsearchNode(&esNode, updateProbingPeriod)
-					}(node)
-
-				}
-				sem.Wait()
-
-				log.Debug("Starting probing Kibana nodes")
-				for _, node := range kibanaNodesList {
-					sem.Add(1)
-					go func(kibanaNode esnode) {
-						defer sem.Done()
-						probeKibanaNode(&kibanaNode, updateProbingPeriod)
-					}(node)
-
-				}
-				sem.Wait()
-			}
-		}
-	},
+type ServeCmd struct {
+	ConsulApi                                string        `default:"127.0.0.1:8500" help:"127.0.0.1:8500" help:"consul target api host:port" short:"a"`
+	ConsulPeriod                             time.Duration `default:"120s" help:"nodes discovery update interval"`
+	ProbePeriod                              time.Duration `default:"30s" help:"elasticsearch nodes probing interval"`
```

Comment:
Can you explicit the purpose of this field in comparison to LatencyProbeRatePerMin ?